nengo_bones/scripts/generate_bones.py

```diff
@@ -1,49 +1,28 @@
 """Scripts for auto-generating nengo-bones files."""
 
-from collections import OrderedDict
 import os
-import stat
 
 import click
-import jinja2
 
-import nengo_bones
+from nengo_bones import __version__
+from nengo_bones.config import flatten_config, load_config, sections
+from nengo_bones.templates import BonesTemplate
 
 
-def render_template(ctx, cfg_key, output_file, template_file=None, **kwargs):
+def render_template(ctx, template):
     """
     Render a template to file.
 
     Parameters
     ----------
     ctx : `click.Context`
         CLI context, containing information specified upstream.
-    cfg_key : str
-        The heading for the section in the config file containing config
-        options specific to the template being rendered.
-    output_file : str
-        Filename for the rendered output file.
-    template_file
-        Filename for the input template file (default is
-        ``output_file + ".template"``)
-    kwargs : dict
-        Will be passed on to the ``render`` function.
+    template : `.BonesTemplate`
+        A `.BonesTemplate` instance representing a templated file.
     """
-    if template_file is None:
-        template_file = output_file + ".template"
-
-    template = ctx.obj["env"].get_template(template_file)
-
-    output_file = os.path.join(ctx.obj["output_dir"], output_file)
-    with open(output_file, "w") as f:
-        f.write(template.render(
-            version=nengo_bones.__version__,
-            **kwargs,
-            # pass in the top-level config options as well
-            # TODO: separate "top-level" config into its own section?
-            **ctx.obj["config"],
-            **ctx.obj["config"][cfg_key],
-        ))
+    kwargs = flatten_config(ctx.obj["config"], template.section)
+    kwargs["version"] = __version__
```

Comment:
This is already set in `render_to_file`

Reply:
I don't think it is set in `render_to_file`? It's a moot point anyhow since it gets removed, but I'm pretty sure it only gets set here.

Reply:
You're right, I'm not sure what I was looking at 🤷‍♂